jaeger-core/src/main/java/io/jaegertracing/internal/MDcScopeManager.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing.internal;
+
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+import io.opentracing.util.ThreadLocalScopeManager;
+
+public class MDcScopeManager implements ScopeManager {
+  private final ScopeManager wrapped = new ThreadLocalScopeManager();
```

Comment:
I don't think it's necessary to assume the concrete implementation of the wrapped scope manager, it can be accepted as ctor argument.

Reply:
done